Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/repair.h

```diff
@@ -1281,17 +1562,45 @@ std::size_t remove_degenerate_faces(TriangleMesh& tmesh,
   return nb_deg_faces;
 }
 
-
 template<class TriangleMesh>
 std::size_t remove_degenerate_faces(TriangleMesh& tmesh)
 {
   return remove_degenerate_faces(tmesh,
     CGAL::Polygon_mesh_processing::parameters::all_default());
 }
 
-template <class TriangleMesh, class Vpm>
-std::size_t duplicate_non_manifold_vertices(TriangleMesh& tm, Vpm vpm)
+/// \ingroup PMP_repairing_grp
+/// duplicates all non-manifold vertices of the input mesh.
+///
+/// @tparam TriangleMesh a model of `FaceListGraph` and `MutableFaceGraph`
+/// @tparam NamedParameters a sequence of \ref pmp_namedparameters "Named Parameters"
+///
+/// @param tm the triangulated surface mesh to be repaired
+/// @param np optional \ref pmp_namedparameters "Named Parameters" described below
+///
+/// \cgalNamedParamsBegin
+///    \cgalParamBegin{vertex_point_map} the property map with the points associated to the vertices of `pmesh`. The type of this map is model of `ReadWritePropertyMap`.
+/// If this parameter is omitted, an internal property map for
+/// `CGAL::vertex_point_t` should be available in `PolygonMesh`
+///    \cgalParamEnd
+///    \cgalParamBegin{geom_traits} a geometric traits class instance.
+///    \cgalParamEnd
+/// \cgalNamedParamsEnd
+///
+/// \return true if the triangle face is degenerate
```

Comment:
I think it might be useful to be able to mark non-manifold vertices, if `vertex_is_constrained_map is passed` for example. Maybe also a way to get the correspondance between those vertices.  For example a new named parameter could take an output iterator where vector of vertex could be put (one vector per group of vertices that were corresponding to the same vertex).